src/core/library.c

```diff
@@ -889,6 +894,75 @@ QuicLibrarySetGlobalParam(
 
         break;
 
+    case QUIC_PARAM_GLOBAL_RAW_DATAPATH_PROCS: {
+        if (Buffer == NULL) {
+            if (MsQuicLib.RawDataPathProcList != NULL) {
+                CXPLAT_FREE(MsQuicLib.RawDataPathProcList, QUIC_POOL_RAW_DATAPATH_PROCS);
+                MsQuicLib.RawDataPathProcList = NULL;
+                MsQuicLib.RawDataPathProcListLength = 0;
+            }
+            Status = QUIC_STATUS_SUCCESS;
+            break;
+        }
+
+        if (BufferLength < sizeof(uint16_t) || BufferLength % sizeof(uint16_t) != 0) {
+            Status = QUIC_STATUS_INVALID_PARAMETER;
+            break;
+        }
+
+        if (MsQuicLib.Datapath != NULL) {
+            QuicTraceLogError(
+                LibraryRawDataPathProcsSetAfterDataPathInit,
+                "[ lib] Tried to change raw datapath procs after datapath initialization");
+            Status = QUIC_STATUS_INVALID_STATE;
+            break;
+        }
+
+        uint32_t RawDataPathProcListLength = BufferLength / sizeof(uint16_t);
+        uint16_t* Cpus = (uint16_t*)Buffer;
+        _Analysis_assume_(BufferLength >= 4); // SAL incorrectly thinks 4 bytes will be accessed.
+        for (uint32_t i = 0; i < RawDataPathProcListLength; ++i) {
+            if (Cpus[i] >= CxPlatProcActiveCount()) {
+                Status = QUIC_STATUS_INVALID_PARAMETER;
+                break;
+            }
+        }
+
+        if (Status == QUIC_STATUS_INVALID_PARAMETER) {
+            QuicTraceLogError(
+                LibraryRawDataPathInvalidProcs,
+                "[ lib] Tried to set invalid raw datapath procs");
+            break;
+        }
+
+        if (MsQuicLib.RawDataPathProcList != NULL) {
```

Comment:
Probably better to only do this if the alloc after succeeds, so there are no unintended side effects in OOM case.